airline-data/src/main/scala/com/patson/data/AirlineSource.scala

```diff
@@ -165,6 +165,15 @@ object AirlineSource {
   	      updateStatement.setInt(2, airlineId)
   	      updateStatement.executeUpdate()
   	      updateStatement.close()
+
+          // update balance audit table
```

Comment:
I'm kinda on the fence for this, in a way it's nice that :
1. It has the potentials to keep entries on every single `adjustAirlineBalance` call and not just the final balance at the end of the cycle. But this would require changing the schema - no longer be `UNIQUE KEY `balance_audit_UN` (`airline`,`cycle`),` otherwise it's kinda meaningless, as entries just overwrite eachother and at the end it will be overwritten by the simulation on airline at cycle end


But the cons are:
1. This is not the only place that set balance (though the most common)
2. With the current schema it's kinda meaningless to write to DB for all the updates during the week by user action, since at the end it will just be replaced by the `AirlineSimulation` before the week ticks

I lean towards setting it in the `AirlineSimulation` instead (and the purge too) - somewhere [here](https://github.com/patsonluk/airline/blob/master/airline-data/src/main/scala/com/patson/AirlineSimulation.scala#L406) , as for cash flow during the week, the `cash_flow_item` table is supposed to keep track of it.